src/org/stringtemplate/v4/misc/ErrorManager.java

```diff
@@ -156,4 +149,14 @@ public void IOError(ST self, ErrorType error, Throwable e, Object arg) {
     public void internalError(ST self, String msg, Throwable e) {
         listener.internalError(new STMessage(ErrorType.INTERNAL_ERROR, self, e, msg));
     }
+
+    private String sourceName(Token t) {
+        CharStream input = t.getInputStream();
+        String srcName = null;
+        if ( input!=null ) {
+            srcName = input.getSourceName();
+            if ( srcName!=null ) srcName = Misc.getFileName(srcName);
+        }
+        return srcName;
```

Comment:
Overall I like the refactoring, especially the deduplication! However, I suggest using early return as well as braces for all if statements:

```suggestion
        CharStream input = t.getInputStream();
        if ( input==null ) {
            return null;
        }
        String srcName = input.getSourceName();
        if ( srcName!=null ) {
            srcName = Misc.getFileName(srcName);
        }
        return srcName;
```